pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -27,13 +27,64 @@ namespace {
 using namespace Pennylane::Util;
 using Kokkos::kokkos_swap;
 using Pennylane::Gates::GateOperation;
+using Pennylane::LightningKokkos::Util::ControlBitPatterns;
+using Pennylane::LightningKokkos::Util::generateBitPatterns;
+using Pennylane::LightningKokkos::Util::parity_2_offset;
+using Pennylane::LightningKokkos::Util::reverseWires;
 using Pennylane::LightningKokkos::Util::vector2view;
 } // namespace
 /// @endcond
 
 namespace Pennylane::LightningKokkos::Functors {
 
-template <class PrecisionT, class FuncT> class applyNC1Functor {
+template <class PrecisionT, class FuncT, bool has_controls = true>
+class applyNC1Functor {};
+
+template <class PrecisionT, class FuncT>
+class applyNC1Functor<PrecisionT, FuncT, true> {
+    using KokkosIntVector = Kokkos::View<std::size_t *>;
+
+    Kokkos::View<Kokkos::complex<PrecisionT> *> arr;
+    const FuncT core_function;
+    KokkosIntVector indices;
+    KokkosIntVector parity;
+    KokkosIntVector rev_wires;
+    KokkosIntVector rev_wire_shifts;
+
+  public:
+    template <class ExecutionSpace>
+    applyNC1Functor([[maybe_unused]] ExecutionSpace exec,
```

Comment:
So this `ExecutionSpace` is passed from https://github.com/PennyLaneAI/pennylane-lightning/blob/5f70b06e286447afb6703ce98df05d379108e83f/pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp#L77 , which will depend on what device execution space Kokkos is built for

Reply:
the argument `exec` is not actually used in the body, so it will complain if we remove `[[maybe_unused]]`. The argument is needed just to infer the class type for `ExecutionSpace`